### ISSUE TITLE
Cannot add MyAnimeList tracking 

### ISSUE THREAD

**Issue by N0153** [mihonapp/mihon#298]
### Steps to reproduce

1. Go to Settings -> Tracking
2. Click on MyAnimeList
3. Log in to MyAnimeList
4. Click on the Allow button
5. (Optional) Use Fingerprint/Insert PIN after being redirected to Mihon

### Expected behavior

The MyAnimeList tracker should be successfully registered to Mihon (the green checkmark should be visible next to the MyAnimeList tracker).

### Actual behavior

The MyAnimeList tracker is not registered to Mihon (no green checkmark appears next to the MyAnimeList tracker).

### Mihon version

0.16.3

### Android version

Android 13

### Device

Asus ZenFone 9 (ASUS_AI2202)

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by Nachogao** [comment]
I has the same problem, but I resolved it closing the app, and doing it again. And now I cannot reproduce the error.

**Comment by N0153** [comment]
> I has the same problem, but I resolved it closing the app, and doing it again. And now I cannot reproduce the error.

After trying multiple times, I finally managed to get the green checkmark on MyAnimeList. 
The problem is that, when I try to update an entry I get a "HTTP error 403" message.

**Comment by vitorggois** [comment]
> > I has the same problem, but I resolved it closing the app, and doing it again. And now I cannot reproduce the error.
> 
> After trying multiple times, I finally managed to get the green checkmark on MyAnimeList. The problem is that, when I try to update any entry I get a "HTTP error 403" message, ![Screenshot_20240131-225527_Mihon.jpg](https://private-user-images.githubusercontent.com/12370033/301320957-73d09112-5449-4edd-be51-193ff95f8447.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3Mzk2MzgsIm5iZiI6MTcwNjczOTMzOCwicGF0aCI6Ii8xMjM3MDAzMy8zMDEzMjA5NTctNzNkMDkxMTItNTQ0OS00ZWRkLWJlNTEtMTkzZmY5NWY4NDQ3LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDIyMTUzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmNTViZGJiNDQwNGJlODMzYjM0NzgwNzBhMDE4YmE0ZGNmN2JkMDBkZmVmOGJhODA3MmQzOGE2MTE5NDhiMmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xUrAD2DrdVIgcG3xGEPvAF4OOZigA39zu5BJsHPCjLs) followed by an "HTTP error 503" message shortly after.![Screenshot_20240131-225637_Mihon.jpg](https://private-user-images.githubusercontent.com/12370033/301321023-8c2ba9f3-e05a-4937-b0fb-576f4a9a0013.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3Mzk2MzgsIm5iZiI6MTcwNjczOTMzOCwicGF0aCI6Ii8xMjM3MDAzMy8zMDEzMjEwMjMtOGMyYmE5ZjMtZTA1YS00OTM3LWIwZmItNTc2ZjRhOWEwMDEzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMxVDIyMTUzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlMTkwM2EyMTgzMWJmNzFjYTJiNTE2MmIwNjFkYWFiMjY3MTc4YmUyYzM3ZjQxODk1NDE3ZGI2YTZjNGYyNWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ODft2t0nmJYAgg7Qlzf9cqgFZuacg1dyC-5eLxCjWMQ)
> 

I'm having the same issue with HTTP errors

**Comment by loocool2** [comment]
![image](https://github.com/mihonapp/mihon/assets/36128021/394d0654-fc97-4b57-99f5-4934cd652f6d)
MAL is still blocking the manga endpoints in the MAL api as of about 12h ago

**Comment by N0153** [comment]
> ![image](https://github.com/mihonapp/mihon/assets/36128021/394d0654-fc97-4b57-99f5-4934cd652f6d)
> MAL is still blocking the manga endpoints in the MAL api as of about 12h ago

Thanks! I cannot access the MAL Discord at the moment. Are there any updates on this issue?

By the way, a few minutes ago I tried to update the tracking on some manga again. This time I didn't get an error (neither a 403 nor a 503), but the tracking still didn't update.

**Comment by loocool2** [comment]
No change since that

**Comment by ultrabear** [comment]
getting the same issue as originally stated too

**Comment by xifutang** [comment]
Thought it was just me lol

**Comment by MajorTanya** [comment]
Thank you all for the reports but please refrain from adding things like "me too" as comments. If you want to signal you're also affected by this, just click the 👍 emoji on the original issue at the top. A lot of people get emails from issues and getting one just to see "happened to me as well" is not the vibe.
Again, thanks for the reports but stick to the emoji reaction 👍

**Comment by dgw** [comment]
Comment 38 in a related thread on MAL's forum indicates that while the related API functionality seems to be restored, Mihon itself is potentially blocked by user-agent. https://myanimelist.net/forum/?goto=post&topicid=2143705&id=70563666

Presumably that would need a Mihon developer to reach out and see what (if anything) MAL needs to unblock it.

**Comment by etreff** [comment]
> Comment 38 in a related thread on MAL's forum indicates that while the related API functionality seems to be restored, Mihon itself is potentially blocked by user-agent. https://myanimelist.net/forum/?goto=post&topicid=2143705&id=70563666
> 
> Presumably that would need a Mihon developer to reach out and see what (if anything) MAL needs to unblock it.

MAL is notorious for being extremely difficult to work with, both from a user perspective and a developer perspective. Mihon would absolutely need to reach out to MAL to resolve this, but I do not expect much. One thing to keep in mind is that while MAL is a tracker they also sell digital copies of licensed manga. They probably wouldn't be to keen on working with an app that easily enables sidestepping their business.

**Comment by Atsuya12** [comment]
Guys, I don't know if this changed to someone else, but my MAL tracking is working again. I'm currently running the Beta r6501 version of the app

**Comment by MajorTanya** [comment]
> Guys, I don't know if this changed to someone else, but my MAL tracking is working again. I'm currently running the Beta r6501 version of the app

We're aware of this, but there are still some blockers on the main version.

**Comment by N0153** [comment]
> > Guys, I don't know if this changed to someone else, but my MAL tracking is working again. I'm currently running the Beta r6501 version of the app
> 
> We're aware of this, but there are still some blockers on the main version.

I can confirm this. I just unregistered the MAL tracker (it wasn't working anyway) and tried to add it again from scratch, but it seems I'm back to square one, as I couldn't even register the tracker with Mihon (no green tick next to the tracker name).

**Comment by KyunnDayo** [comment]
02/16/24
running mihon stable 0.16.3

myanimelist tracking seems to be working now

**Comment by N0153** [comment]
> 02/16/24
> running mihon stable 0.16.3
> 
> myanimelist tracking seems to be working now

I just checked it myself. I can also confirm that it is working fine at the moment.

**Comment by SoumickDeyArgha** [comment]
Using Mihon stable 0.16.3

> 02/16/24 running mihon stable 0.16.3
> 
> myanimelist tracking seems to be working now

I have tried everything from the scratch, but still its not working for me

**Comment by dgw** [comment]
That forum thread I linked [before](https://github.com/mihonapp/mihon/issues/298#issuecomment-1936446072) is active again with more problem reports.

I really hope the MAL team will make a statement about this, but if they are targeting specific clients/UAs then it's probably intentional.

**Comment by MajorTanya** [comment]
Only reopening so people find this easier, I don't want to see any "+1" or "me too" or "when is this getting fixed???!1!??"

We know as much as you do when it comes to the reasoning behind this.

**Comment by Th3-822** [comment]
maybe there are lots of users retrying as while it's blocked it justs gets timeout on mihon instead of returning a real error code, like 403, 429 or 503 while its blocked (it returns 504 iirc after 120 seconds and mihon doesn't wait that much for the reply)

increasing the timeout and maybe adding a forced logout if there is a error updating to just stop sending invalid requests as maybe there are some users still using expired/invalid tokens (from "old" backups) without noticing it

or just allowing to customize the useragent for the trackers _(and maybe api's appid & public/secret tokens too)_ to be able to avoid any future block

**Comment by MajorTanya** [comment]
> maybe there are lots of users retrying as while it's blocked it justs gets timeout on mihon instead of returning a real error code, like 403, 429 or 503 while its blocked (it returns 504 iirc after 120 seconds and mihon doesn't wait that much for the reply)
> 
> increasing the timeout and maybe adding a forced logout if there is a error updating to just stop sending invalid requests as maybe there are some users still using expired/invalid tokens (from "old" backups) without noticing it
> 
> or just allowing to customize the useragent for the trackers _(and maybe api's appid & public/secret tokens too)_ to be able to avoid any future block

- 504 is literally just a timeout message (meaning the server didn't respond at all), and no well-behaved API takes longer than Mihon is waiting, especially not 120 seconds/2 full minutes. (see below for an educational illustration)
- Allowing for UA replacements for trackers would be a pretty fundamental change to the structure of the code, and I don't know if it would even be API Licence/ToS compliant to allow for it in the first place
- Users should stop trying if the tracker isn't responding
- Further optimisations (& suggestions) on Mihon's end to maybe curtail some of the issues are always welcome, but no viable ideas have been floated thus far, and MAL doesn't appear to be communicating with anyone, let alone Mihon

![http://http.cat/504](http://http.cat/504)

**Comment by breddie-normie** [comment]
Recently migrated from Tachiyomi to Mihon. I was able to get the MAL token, but I noticed that MAL isn't tracking the progress. On further fiddling, I noticed that anything MAL related isn't working with Mihon. Now I tried re-enabling the token but now I get this "400 Bad Request" error. wtf is MAL doing with these apps, didn't Mihon specifically remove pre-configured extension list to avoid legal issues?

**Comment by BraveSoldat** [comment]
> Recently migrated from Tachiyomi to Mihon. I was able to get the MAL token, but I noticed that MAL isn't tracking the progress. On further fiddling, I noticed that anything MAL related isn't working with Mihon. Now I tried re-enabling the token but now I get this "400 Bad Request" error. wtf is MAL doing with these apps, didn't Mihon specifically remove pre-configured extension list to avoid legal issues?

I don't think this is limited to Mihon.
I use PocketMAL for manually tracking anime and physical manga I read and I had some trouble with it when this API problem began.
Sometimes PocketMAL still has some connection issues with MAL, so it isn't really normalized yet.
I guess MAL began monitoring their API usage requests more closely and are trying to make third parties lower the rate of requests. Hence the API usage notice Mihon devs received from MAL.
If devs don't comply, then MAL resorts to more drastic measures.

That's just my guess though.

**Comment by breddie-normie** [comment]
> > Recently migrated from Tachiyomi to Mihon. I was able to get the MAL token, but I noticed that MAL isn't tracking the progress. On further fiddling, I noticed that anything MAL related isn't working with Mihon. Now I tried re-enabling the token but now I get this "400 Bad Request" error. wtf is MAL doing with these apps, didn't Mihon specifically remove pre-configured extension list to avoid legal issues?
> 
> I don't think this is limited to Mihon. I use PocketMAL for manually tracking anime and physical manga I read and I had some trouble with it when this API problem began. Sometimes PocketMAL still has some connection issues with MAL, so it isn't really normalized yet. I guess MAL began monitoring their API usage requests more closely and are trying to make third parties lower the rate of requests. Hence the API usage notice Mihon devs received from MAL. If devs don't comply, then MAL resorts to more drastic measures.
> 
> That's just my guess though.

Thanks man. Sorry for being a dick. For one thing, I know MAL isn't actually very good in server stability department, but trying to restrict API calls is scummy man. If MAL wants people to use their shitass app, they should make it function better than their FOSS counterparts. If shit starts hitting the ceiling, I think it's time to consider Anilist (let it be however I loved MAL's classic image board style forums).

**Comment by Appoxo** [comment]
Juat tried it with Tachiyomi (0.15.3) and Mihon (0.16.3) on Pixel 7 Pro (Android 14)

Tachiyomi worked first try
Mihon loaded forever

**Comment by SoumickDeyArgha** [comment]
Tried with Tachiyomi today
I did changes the user agent and it's work like a charm.

`Mozilla/5.0 (Linux; Android 13) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/20.0 Chrome/106.0.5249.126 Mobile Safari/537.36`

**Comment by Appoxo** [comment]
> Tried with Tachiyomi today
> I did changes the user agent and it's work like a charm.
> 
> `Mozilla/5.0 (Linux; Android 13) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/20.0 Chrome/106.0.5249.126 Mobile Safari/537.36`

You mean you changed the Mihon user agent or the tachiyomi one?

**Comment by SoumickDeyArgha** [comment]
> > Tried with Tachiyomi today
> > I did changes the user agent and it's work like a charm.
> > `Mozilla/5.0 (Linux; Android 13) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/20.0 Chrome/106.0.5249.126 Mobile Safari/537.36`
> 
> You mean you changed the Mihon user agent or the tachiyomi one?

Tachiyomi Worked with chage User agent
Mihon didn't I tried different user agent but nothing working

**Comment by SoumickDeyArgha** [comment]
> They were specifically blocking Mihon for the past couple days, but from what I've been seeing tonight, their API might be down again

If that's then case, then some user mentioned the Mal tracking working in Mihon preview, idk what could be thr reason behind it

**Comment by MajorTanya** [comment]
Users can't influence the User-Agent Mihon uses for MAL. And I'm 90% sure the preview doesn't work either because of how MAL is blocking Mihon.

**Comment by Appoxo** [comment]
> > They were specifically blocking Mihon for the past couple days, but from what I've been seeing tonight, their API might be down again
> 
> If that's then case, then some user mentioned the Mal tracking working in Mihon preview, idk what could be thr reason behind it

Tried it yesterday with the newest release. Either it doesnt work or I was too impatient.

**Comment by MajorTanya** [comment]
> > > They were specifically blocking Mihon for the past couple days, but from what I've been seeing tonight, their API might be down again
> > 
> > If that's then case, then some user mentioned the Mal tracking working in Mihon preview, idk what could be thr reason behind it
> 
> Tried it yesterday with the newest release. Either it doesnt work or I was too impatient.

It doesn't work at all for any Mihons right now

**Comment by dgw** [comment]
7974a1fc0c2b0e237d5704a033a60ec0fd60cdbc being included in 0.16.4 released earlier today sounded like it would make MAL auth/tracking work again, but I'm still getting timeouts. Is there another step to this besides updating the Mihon app? Maybe a cache to clear?

**Comment by Fuuzetsu** [comment]
I also confirm no improvement on r6530, just getting the spinner.

**Comment by dgw** [comment]
@Smol-Ame Thanks, that was the missing step. Made a few tweaks to the UA string and I was able to add MAL back in the Tracking settings :tada:

**Comment by vitorggois** [comment]
> For the both of you, please change your default user-agent (Mihon Settings > Advanced) to something else. Then it should work

Tried a bunch of user-agents that I found on the troubleshoot link, still got nothing. Anyone can tell me an user-agent that worked?

**Comment by SoumickDeyArgha** [comment]
> > For the both of you, please change your default user-agent (Mihon Settings > Advanced) to something else. Then it should work
> 
> Tried a bunch of user-agents that I found on the troubleshoot link, still got nothing. Anyone can tell me an user-agent that worked?

You can try this one
`Mozilla/5.0 (Linux; Android 13) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/20.0 Chrome/106.0.5249.126 Mobile Safari/537.36`

**Comment by Appoxo** [comment]
> > For the both of you, please change your default user-agent (Mihon Settings > Advanced) to something else. Then it should work
> 
> Tried a bunch of user-agents that I found on the troubleshoot link, still got nothing. Anyone can tell me an user-agent that worked?

If I try tobauth, Mihin open the native Firefix App. So idk if it matters.
Anyway, heres mine: 
`Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/120.0`

**Comment by BraveSoldat** [comment]
It's working for me with this UA: 

Mozilla/5.0 (Linux; Android 13) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/20.0 Chrome/106.0.5249.126 Mobile Safari/537.36

**Comment by vitorggois** [comment]
Reinstalled the whole app, seems to be working now.

**Comment by Doministo** [comment]
> > > For the both of you, please change your default user-agent (Mihon Settings > Advanced) to something else. Then it should work
> > 
> > 
> > Tried a bunch of user-agents that I found on the troubleshoot link, still got nothing. Anyone can tell me an user-agent that worked?
> 
> If I try tobauth, Mihin open the native Firefix App. So idk if it matters. Anyway, heres mine: `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/120.0`

To be honest I just changed `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/121.0` to `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/120.0` and then it worked, so nice and thanks yall
